### PR TITLE
CMTS-19: Mule Agent Plugin for MTS Services

### PR DESCRIPTION
**Pull request by lucianoRM**

### CODE REVIEW

File: modules/tooling-support-parent/tooling-support/pom.xml [hunk 1]
```diff
@@ -43,6 +42,11 @@
             <artifactId>mule-module-deployment</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>29.0-jre</version>
```
Comment (gsfernandes) [inline modules/tooling-support-parent/tooling-support/pom.xml:48]: Check if this version is already managed by Mule's pom.xml

File: modules/tooling-support-parent/tooling-support/src/main/java/org/mule/runtime/module/tooling/api/ToolingService.java [hunk 2]
```diff
@@ -38,6 +39,8 @@
    */
   ConnectivityTestingServiceBuilder newConnectivityTestingServiceBuilder();
 
+  DeclarationSessionBuilder newDeclarationSessionBuilder();
```
Comment (gsfernandes) [inline modules/tooling-support-parent/tooling-support/src/main/java/org/mule/runtime/module/tooling/api/ToolingService.java:42]: Javadoc

File: modules/tooling-support-parent/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/config/DefaultDeclarationSession.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  private LazyValue<DeclarationSession> internalConfigurationService;
+
+  DefaultDeclarationSession(ApplicationSupplier applicationSupplier) {
+    super(applicationSupplier);
+    this.internalConfigurationService = new LazyValue<>(() -> {
+      try {
+        return createInternalService(getStartedApplication());
+      } catch (ApplicationStartingException e) {
+        throw new MuleRuntimeException(e);
```
Comment (gsfernandes) [inline modules/tooling-support-parent/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/config/DefaultDeclarationSession.java:34]: throw new MuleRuntimeException(e.getCause())?

File: modules/tooling-support-parent/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/config/params/EmptyParameterValueResolver.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+package org.mule.runtime.module.tooling.internal.config.params;
+
+import static java.util.Collections.emptyMap;
+import org.mule.runtime.module.extension.internal.runtime.ValueResolvingException;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ParameterValueResolver;
+import org.mule.runtime.module.extension.internal.runtime.resolver.ValueResolver;
+
+import java.util.Map;
+
+public class EmptyParameterValueResolver implements ParameterValueResolver {
```
Comment (gsfernandes) [inline modules/tooling-support-parent/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/config/params/EmptyParameterValueResolver.java:16]: Not used, remove this class.

File: modules/tooling-support-parent/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/config/cache/NoOpMetadataCache.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.metadata.MetadataCache;
+import org.mule.runtime.api.metadata.MetadataResolvingException;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Optional;
+
+public class NoOpMetadataCache implements MetadataCache {
```
Comment (gsfernandes) [inline modules/tooling-support-parent/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/config/cache/NoOpMetadataCache.java:17]: Not used, remove this class. It would be added as part of Metadata support..

**Comment by lucianoRM** [comment]
--validate

**Comment by gsfernandes** [comment]
--validate

**Comment by lucianoRM** [comment]
--validate

**Comment by lucianoRM** [comment]
--validate

**Comment by gsfernandes** [comment]
--validate

**Comment by lucianoRM** [comment]
--validate

**Comment by gsfernandes** [comment]
--validate

**Comment by facunlk** [comment]
--validate

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationParameterValueResolver.java [hunk 6]
```diff
@@ -41,10 +41,10 @@
   private final Map<String, String> showInDslParameters;
   private final ReflectionCache reflectionCache;
 
-  OperationParameterValueResolver(ExecutionContext<T> executionContext,
-                                  ResolverSet resolverSet,
-                                  ReflectionCache reflectionCache,
-                                  ExpressionManager expressionManager) {
+  public OperationParameterValueResolver(ExecutionContext<T> executionContext,
```
Comment (gsfernandes) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationParameterValueResolver.java:44]: Revert this change, seems to be not needed.

**Comment by lucianoRM** [comment]
--validate

**Comment by gsfernandes** [comment]
--validate

**Comment by gsfernandes** [comment]
--
